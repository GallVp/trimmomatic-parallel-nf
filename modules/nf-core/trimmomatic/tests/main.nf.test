nextflow_process {

    name "Test Process TRIMMOMATIC"
    script "../main.nf"
    process "TRIMMOMATIC"
    tag "modules"
    tag "modules_nfcore"
    tag "trimmomatic"

    test("Paired-Reads") {
        config "./nextflow_PE.config"
        when {
            process {
                """
                input[0] = [
                    [ id: 'test', single_end:false ],
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                    ]
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert path(process.out.trimmed_reads[0][1][0]).fastq.getNumberOfRecords() == 86 },
                { assert path(process.out.trimmed_reads[0][1][1]).fastq.getNumberOfRecords() == 86 },
                { assert snapshot(
                    process.out.versions,
                    [
                        "R1",
                        java.security.MessageDigest.getInstance("MD5").digest(
                            path(process.out.trimmed_reads[0][1][0]).fastq.sequences.toSorted().join("\n").bytes
                        ).collect { String.format("%02x", it) }.join(""),
                    ],
                    [
                        "R2",
                        java.security.MessageDigest.getInstance("MD5").digest(
                            path(process.out.trimmed_reads[0][1][1]).fastq.sequences.toSorted().join("\n").bytes
                        ).collect { String.format("%02x", it) }.join(""),
                    ]
                    ).match() }
            )
        }
    }

}