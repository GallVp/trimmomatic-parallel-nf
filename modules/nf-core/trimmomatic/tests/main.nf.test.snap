{
    "Paired-Reads": {
        "content": [
            [
                "versions.yml:md5,14413a048f088a147fb04f3d59c6c604"
            ],
            [
                "R1",
                "90fd3bc516b1c2b84d41fdfda2312590"
            ],
            [
                "R2",
                "ccf73b40cb0cc8fbe5a90b35d064735c"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-23T10:11:13.090106"
    }
}